Sources/Extraction/Filters/HillOrientation.c

```diff
@@ -1,26 +1,230 @@
+                printf("strength: %f, orientation: (%f, %f), pixelValue: %f neighborValue: %f, anti-neighborValue: %f\n", strength, orientation.x, orientation.y, pixelValue, neighborValue, antiNeighborValue);
+                printf("Setting hill orientation for %d, %d: (%f, %f)\n", x, y, contribution.x, contribution.y);*/
+                
+                if (strength > 0) {
+                    directions.data[x][y] = Calc_Add2PointsF(&directions.data[x][y], &contribution);
+                }
+            }
+
+            //printf("FINISHED Calculate hill orientation for %d, %d: (%f, %f) ------------------------------------------\n", x, y, directions.data[x][y].x, directions.data[x][y].y);
+
+        }
+    }
+
+    //print_directions(directions);
+    return directions;
+}
+
+PointFArray2D HillOrientation_SumBlocks(const PointFArray2D * pixelOrientations, BinaryMap * blockMask, BlockMap * blocks)
 {
-    HillOrientation ho = {
-        .minHalfDistance = 2,    /* Lower = 0.5, Upper = 4 */
-        .maxHalfDistance = 6,    /* Lower = 5, Upper = 13 */
-        .neighborListSplit = 50, /* Upper = 100 */
-        .neighborsChecked = 20,  /* Upper = 100 */
-        .smoothingRadius = 1     /* Upper = 3 */
-    };
+    const PointFArray2D blockOrientations = PointFArray2D_Construct(blocks->blockCount.width, blocks->blockCount.height);
+    const int blockHeight = blocks->pixelCount.height / blocks->blockCount.height;
+    const int blockWidth = blocks->pixelCount.width / blocks->blockCount.width;
+
+    for (int blockX = 0; blockX < blocks->blockCount.width; blockX++)
+    {
+        for (int blockY = 0; blockY < blocks->blockCount.height; blockY++)
+        {
+            if (!BinaryMap_GetBit(blockMask, blockX, blockY))
+                continue;
 
-    return ho;
+            int bottomLeftX = blocks->blockAreas.corners.allX.data[blockX];
+            int bottomLeftY = blocks->blockAreas.corners.allY.data[blockY];
+            for (int x = bottomLeftX; x < bottomLeftX + blockWidth; ++x) {
+                for ( int y = bottomLeftY; y < bottomLeftY + blockHeight; ++y ) {
+                    blockOrientations.data[blockX][blockY] = Calc_Add2PointsF(&blockOrientations.data[blockX][blockY], &pixelOrientations->data[x][y]);
+                }
+            }
+        }
+    }
+    return blockOrientations;
 }
 
-void HillOrientation_Destruct(HillOrientation *me)
+PointFArray2D HillOrientation_SmoothDirections(PointFArray2D directions, BinaryMap blockMask)
 {
+    // PointFArray2D smoothedDirections = PointFArray2D_Construct()
+    return  (PointFArray2D) { NULL };
 }
 
-void /*UInt8Array2D*/ HillOrientation_Detect(const HillOrientation *me, const FloatArray2D *image, const BinaryMap *mask, const BlockMap *blocks)
+BoolArray2D HillOrientation_BlockMapToPixelMask(Size imageDimensions, BinaryMap * blockMask, BlockMap * blocks) {
```

Comment:
BinaryMap is a more memory efficient version of BoolArray2D. Though CPU performance is more important I expect.